File: backend/routes/trainers.js
/**
 * Trainer Application Routes
 */

const express = require('express');
const router = express.Router();
const { v4: uuidv4 } = require('uuid');
const { addToCollection } = require('../services/database');

// Submit trainer application
router.post('/', (req, res) => {
  const { name, email, phone, expertise, experience, city, message } = req.body;

  // Validation
  if (!name || !email || !expertise) {
    return res.status(400).json({
      error: 'Missing required fields',
      required: ['name', 'email', 'expertise']
    });
  }

  const newApplication = {
    id: uuidv4(),
    name,
    email,
    phone: phone || null,
    expertise,
    experience: experience || null,
    city: city || 'Bengaluru',
    message: message || null,
    createdAt: new Date().toISOString(),
    status: 'pending'
  };

  addToCollection('trainerApplications', newApplication);

  res.status(201).json({
    success: true,
    message: 'Thank you for your application! Our team will review it and get back to you within 48 hours.',
    application: newApplication
  });
});

module.exports = router;

File: backend/routes/trainers.py
# Trainer Routes

from fastapi import APIRouter
from datetime import datetime, timezone
import uuid
from models.schemas import TrainerApplicationCreate
from services.database import add_to_collection

router = APIRouter(prefix="/trainer-applications", tags=["Trainer Applications"])


@router.post("")
async def submit_trainer_application(application: TrainerApplicationCreate):
    """Submit a trainer application"""
    new_application = {
        "id": str(uuid.uuid4()),
        "name": application.name,
        "email": application.email,
        "phone": application.phone,
        "expertise": application.expertise,
        "experience": application.experience,
        "city": application.city,
        "message": application.message,
        "createdAt": datetime.now(timezone.utc).isoformat(),
        "status": "pending"
    }
    
    add_to_collection("trainerApplications", new_application)
    
    return {
        "success": True,
        "message": "Thank you for your application! Our team will review it and get back to you within 48 hours.",
        "application": new_application
    }
